### DISCUSSION TITLE
Feature: Publish client library as NPM package

### DISCUSSION THREAD

**Discussion by ShivamJoker** [umami-software/umami#1074] [Ideas]
It would be great to have the client library as a module so that it can be easily integrated in applications like React and React Native instead of accessing the global version.

We could do like 

```js
import Umami from 'umami-client'

const tracker = new Umami({config params...})
```

I am not able to locate the library in the src code as well.

**Comment by ShivamJoker** [comment]
@mikecao If I write the modular version will you publish it as npm package?

**Reply by birkhofflee** [reply]
I think you can publish your own and people who need it will adopt it into their applications anyway

**Reply by briancao** [reply]
We're currently working on an api-client.  We're still flushing out how we're handling authorization for it.  Likely to be released Q1 next year.

**Reply by birkhofflee** [reply]
> We're currently working on an api-client. We're still flushing out how we're handling authorization for it. Likely to be released Q1 next year.

I'm wondering if there's any news on this?

**Reply by briancao** [reply]
@BirkhoffLee It's been officially released and documented.

https://umami.is/docs/api-client

**Reply by ShivamJoker** [reply]
> @BirkhoffLee It's been officially released and documented.
> 
> https://umami.is/docs/api-client

But it's for NodeJS how do we use it in a Browser or non node environment?

**Reply by briancao** [reply]
That 'node requirement' needs to be clarified as we're using cross-fetch under the hood to make API calls.
